.github/workflows/deploy.yml

```diff
@@ -0,0 +1,42 @@
+name: Deploy
+
+on:
+  push:
+    branches: [main]
+
+concurrency:
```

Comment:
while i was at it i also added concurrency to cancel out previous builds on the same branch/PR to avoid wait times on CI (didn't do this on release as release should ALWAYS run)

Reply:
yep exactly - the `github.workflow` is there to ensure different triggers don't cancel each other out. `head_ref` is available for PRs and `run_id` for everything else